contracts/DomainRegistry.sol

```diff
@@ -23,94 +63,139 @@ contract DomainRegistry is Initializable, OwnableUpgradeable {
-        require(_domains[domainName] == msg.sender, "Domain should be unregistered by the domain owner");
+        if (_domains[domainName] == address(0)) {
+            revert DomainRegistryDomainIsNotRegistered({ domainName: domainName });
+        }
+        if (_domains[domainName] != msg.sender) {
+            revert DomainRegistryOnlyDomainOwnerAllowed({ domainName: domainName });
+        }
         _;
     }
 
+    /// @notice Retrieves the owner of a domain
+    /// @param domainName The domain name to query the owner for
+    /// @return The owner's address of the domain
     function getDomain(string memory domainName) public view returns (address) {
         string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
 
         return _domains[rootDomain];
     }
 
-    function validateDomainRegistration(string memory domainName) internal
-        checkDomainParent(domainName)
-        checkDomainAvailability(domainName)
-        checkSufficientEtn(domainName)
-    {
+    /// @notice Validates domain registration by checking availability, parent existence, and ETN sent
+    /// @param domainName The domain name to validate registration for
+    function validateDomainRegistration(
+        string memory domainName
+    ) internal checkDomainParent(domainName) checkDomainAvailability(domainName) checkSufficientEtn(domainName) {
+        return;
     }
 
+    /// @notice Allows an owner to withdraw their accumulated rewards
     function withdraw() external {
-        uint share = _shares[msg.sender];
+        uint256 share = _shares[msg.sender];
+        if (share == 0) {
+            revert WithdrawNoBalanceAvailable();
+        }
         _shares[msg.sender] = 0;
         payable(msg.sender).transfer(share);
     }
 
-    function registerDomain(string memory domainName) external payable
-        checkDomainLength(domainName)
-    {
+    /// @notice Registers a domain, store balance into shares so can be withdrown later
+    /// @param domainName The domain name to register
+    function registerDomain(string memory domainName) external payable checkDomainLength(domainName) {
         string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
         validateDomainRegistration(rootDomain);
-
+        string memory parentDomain = DomainParserLibrary.getParentDomain(domainName);
+        address parentDomainOwner = _domains[parentDomain];
         uint256 price = getDomainPrice(rootDomain);
 
-        _shares[owner()] += price;
-
-        uint256 excess = msg.value - price;
-        if (excess > 0) {
-            payable(msg.sender).transfer(excess);
+        // parent domain reward
+        uint256 parentReward;
+        if (parentDomainOwner == address(0)) {
+            parentReward = 0;
+        } else {
+            parentReward = (price * REWARD_PERCENT_OWNER) / 100;
         }
+        _shares[parentDomainOwner] = parentReward;
```

Comment:
Why `_shares[parentDomainOwner] = parentReward;` but not `_shares[parentDomainOwner] += parentReward;` to accumulate the ether balance??.  

You can even treat it as security issue as parent domain owner will loose ether. 
-2 points

Reply:
fixed there https://github.com/vpmoon/blockchain-playground/pull/6/commits/e62a7da02a6c5c9b574df92460907eb0a88e93de